client/src/Pages/CreateActivity.jsx

```diff
@@ -0,0 +1,205 @@
+    const [link, setLink] = useState('');
+    const [address, setAddress] = useState('');
+    const [openingHours, setOpeningHours] = useState('');
+    const [remark, setRemark] = useState('');
+    const [date, setDate] = useState('');
+
+    const navigate = useNavigate();
+    const { data } = useFetch(`http://localhost:4000/api/users`);
+    const { id } = useParams();
+    const [loading, setLoading] = useState(false);
+
+    useEffect(() => {
+        const verifyCookie = async () => {
+            if (!cookies.token) {
+              navigate("/login");
+              return;
+            }
+            
+            try {
+                const res = await axios.post("http://localhost:4000", {}, { withCredentials: true });
+
+                setEmail(res.data.email);
+
+                if (!res.data.status) {
+                  removeCookie("token");
+                  navigate("/login");
+                }
+            } catch (error) {
+                console.error('verifyCookie error:', error);
+                navigate("/login");
+            }
+        };
+
+        const getUser = () => {
+            if (email !== "") {
+                try {
+                    if (data.data) {
+                        setUserId(data.data.find(user => user.email === email)._id);
+                    }
+                } catch (error) {
+                    console.error('getUser error:', error);
+                }
+            }
+        }
+
+        const getItinerary = async () => {
+            if (userId !== "") {
+                try {
+                    const res = await axios.get(`http://localhost:4000/itineraries/${userId}`, { withCredentials: true });
+                    const selectedItinerary = res.data.data.find(itinerary => itinerary._id === id);
+                    console.log(selectedItinerary);
+                    if (selectedItinerary) {
+                        setItineraryId(selectedItinerary._id);
+                        console.log(itineraryId);
+                    } else {
+                        console.error('Selected itinerary not found');
+                    }
+                } catch (error) {
+                    console.error('getItinerary error:', error);
+                }
+            }
+        }
+          
+        verifyCookie();
+        getUser();
+        getItinerary();
+    }, [navigate, cookies, removeCookie, email, data, id, userId]);
+
+    const handleError = (err) => {
+        toast.error(err, {
+            position: "bottom-left",
+        });
+    }
+
+    const handleSuccess = (msg) => {
+        toast.success(msg, {
+            position: "bottom-left",
+            autoClose: 3000,
+        });
+    }
+
+    const handleSave = async () => {
+        const data = {
+            title,
+            link,
+            address,
+            openingHours,
+            remark,
+            date
+        };
+        setLoading(true);
+
+        try {
+            if (title === "") {
```

Comment:
i think you might have missed out checking whether the date of the activity falls under the trip so can maybe do a check here